### PR TITLE
test: increase test coverage to 100

### LINKED ISSUE #42: Bring test coverage to 100%
When running npm test, the current coverage is shown

```
-------------------|----------|----------|----------|----------|-------------------|
File               |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
-------------------|----------|----------|----------|----------|-------------------|
All files          |    98.04 |    85.71 |      100 |       98 |                   |
 src               |    96.77 |    82.35 |      100 |    96.77 |                   |
  error-request.ts |     87.5 |    83.33 |      100 |     87.5 |                 6 |
  index.ts         |      100 |    66.67 |      100 |      100 |                11 |
  wrap-request.ts  |      100 |      100 |      100 |      100 |                   |
 test              |      100 |      100 |      100 |      100 |                   |
  octokit.ts       |      100 |      100 |      100 |      100 |                   |
-------------------|----------|----------|----------|----------|-------------------|
```

after running `npm test`, the detailed coverage report can be browsed at `coverage/lcov-report/index.html`.

I'd love to bump test coverage to 100% just like the rest of the `octokit/*.js` repositories. If you'd like to take it on, just comment here and start your pull request. Make sure to update the `"jest"` settings in package.json:

https://github.com/octokit/plugin-retry.js/blob/dbd3b9057665673ca336f78d53ea88d5651ae200/package.json#L47-L57

### PR DESCRIPTION
**Pull request by copperwall**
Hey @gr2m 👋  I took a shot at bringing the test coverage to 100.

This adds three new test cases that cover edge cases in error-request.ts
and index.ts. These in particular cover
* throwing an error if error.request is undefined
* overriding the state.retries property with the options.request.retries
  in errorRequest().
* asserting that the plugin uses the default options when no options
  argument is passed to it.

Fixes #42

### CODE REVIEW

File: test/retry.test.ts [hunk 1]
```diff
@@ -262,4 +266,87 @@ describe("Automatic Retries", function () {
 
     expect(caught).toEqual(testStatuses.length);
   });
+
+  it("Should use default retry options if no options are passed to the plugin", async function () {
```
Comment (copperwall) [inline test/retry.test.ts:270]: One caveat about this test case is that I'm not sure how to shorten the exponential back retry delay, since I can't pass a small `retryAfterBaseValue` to `retry` otherwise it won't be testing the default parameter functionality that I'm trying to cover with this test case.

I set up four requests so that the test asserts that the default number of retries happens (3) before failing, but that means it takes about 14 seconds for the test case to run with the default exponential backoff.

A slightly more unit test-y approach I can take is to call `retry` with an octokit instance and no options and assert that no TypeErrors are thrown due to trying to access properties on an `undefined` options argument inside the `retry` function.

Comment (gr2m) [reply test/retry.test.ts:270]: I'd rather have slow tests than incomplete tests :) Thank you so much for taking that on! I'd say we worry about the coverage first, and then look into making the tests run faster. But maybe it's not a problem, it's a very small library that does one thing, I think the impact of slow tests is neglect-able and well contained within this plugin

**Comment by gr2m** [comment]
the tests seem to be failing because of prettier. Can you fix the formatting?

**Comment by copperwall** [comment]
oh oops! I could've sworn I ran lint:fix, I must've missed that on the last few changes

**Review by gr2m — APPROVED** [review]
Great work, thank you :)
